docs/tutorials/getting-started.md

```diff
@@ -0,0 +1,138 @@
+# Getting started
+
+`haskell.nix` can automatically translate your Cabal or Stack project and its dependencies into Nix code.
+
+This tutorial will teach you how to setup your project
+
+Assuming you have [Nix installed](https://nixos.org/download.html), you can start setting up your project.
+
+## Setting up the Cachix binary cache
+
+You can **avoid compiling GHC and nix-tools** by configuring [Cachix](https://cachix.org)
+so you can benefit from the binary cache built by CI:
+
+```bash
+$ nix-env -iA cachix -f https://cachix.org/api/v1/install
+$ cachix use iohk
+```
+
+## Scaffolding
+
+The following configuration will use `stack.yaml` if it exists,
+otherwise fallback to `cabal.project`.
```

Comment:
I'm not sure about this: nobody actually wants this (they have one or the other or they'll pick one), and it actually makes the example more complicated since we have this file existence testing stuff.

I think it's fine to just have two examples.

Reply:
I think that it's better to give one example to copy and people can edit out their own workflow.

Most important feature is that scaffold gives a good default.

Just a few days ago I fixed https://github.com/input-output-hk/haskell.nix/commit/2793871216a80e485ec1a78a4117f5822cce5b8d because the two templates weren't in sync.

Also having to split the flow during a getting started is considered bad practice in documentation.

We could alternatively offer two snippets that user can comment/uncomment, but that means copy/pasting just won't work out of the box.

There are tradeoffs here, but I think saying prefer Stack over Cabal is a good default.

Reply:
> Stack over Cabal

You might have come to the wrong repo 😈; especially with Stack's pantry stuff, which we simply can't support this is a dangerous direction to go as people will expect Stack to be the preferred tool of choice. For the vast majority of iohk that's simply not the case, and thus the likelihood of Cabal fixes will be much higher than for Stack. Simply because we hardly use Stack at all, and thus won't see much of Stack breakage outside of a selected few repositories. We sadly rely on the community to fix Stack to a large degree 😢